src/pages/JoinStep2/JoinStep2.js

```diff
@@ -0,0 +1,365 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import "./JoinStep2.scss";
+
+class JoinStep2 extends Component {
+  constructor() {
+    super();
+    this.state = {
+      name: "",
+      account: "",
+      password: "",
+      email: "",
+      phoneNumber: "",
+      dateOfBirth: 19951116,
+      emailAgreement: false,
+      smsAgreement: false,
+      //여기서부터 백에 전달 안함
+      confirmPw: "",
+      isNameNull: true,
+      isAccountNull: true,
+      isPasswordNull: true,
+      isEmailNull: true,
+      isPhoneNumberNull: true
+    };
+  }
+
+  handleInput = e => {
+    e.preventDefault();
+    const { value, name } = e.target;
+    this.setState({
+      [name]: String(value)
+    });
+    console.log(this.state);
+  };
+
+  handleSMSCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.smsAgreement
+    });
+    console.log(this.state);
+  };
+
+  handleEmailCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.emailAgreement
+    });
+    console.log(this.state);
+  };
+
+  checkId = () => {
+    const APIforIdCheck = "http://localhost:3000/data/account.json";
+    const { account } = this.state;
+    fetch(APIforIdCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        account: account
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  checkEmail = () => {
+    const APIforEmailCheck = "http://localhost:3000/data/email.json";
+    const { email } = this.state;
+    fetch(APIforEmailCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        email: email
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  isPwInfoValid = password => {
+    let number = /[0-9]/g;
+    let character = /[a-zA-z]/g;
+    let specialCharacter = /\W|_/g;
+    const isNumIncluded = password.match(number) ? true : false;
+    const isCharIncluded = password.match(character) ? true : false;
+    const isSpecialCharIncluded = password.match(specialCharacter)
+      ? true
+      : false;
+    const isPwLong = password.length >= 6 && password.length <= 16;
+    if (
+      ((isNumIncluded && isSpecialCharIncluded) ||
+        (isNumIncluded && isCharIncluded) ||
+        (isCharIncluded && isSpecialCharIncluded)) &&
+      isPwLong
+    ) {
+      return true;
+    } else return false;
+  };
+
+  isInputNull = () => {
+    const { name, account, password, email, phoneNumber } = this.state;
+    name === ""
+      ? this.setState({ isNameNull: false })
+      : this.setState({ isNameNull: true });
```

Comment:
```suggestion
      this.setState({ isNameNull: name !== "" })
```
setState를 두번 쓰지 않아도 될 것 같습니다. 왜냐면 비교한 결과가 boolean으로 리턴되기 때문이죠!